### PR TITLE
Add type hints for XLM TensorFlow

### PR DESCRIPTION
**Pull request by elusenji**
# What does this PR do?
This PR adds type hints to the XLM model for TensorFlow.

@LysandreJik

### CODE REVIEW

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 1]
```diff
@@ -558,8 +559,8 @@ class TFXLMWithLMHeadModelOutput(ModelOutput):
     """
 
     logits: tf.Tensor = None
-    hidden_states: Optional[Tuple[tf.Tensor]] = None
-    attentions: Optional[Tuple[tf.Tensor]] = None
+    hidden_states: Optional[tf.Tensor] = None
+    attentions: Optional[tf.Tensor] = None
```
Comment (kumar-abhishek) [inline src/transformers/models/xlm/modeling_tf_xlm.py:563]: Is it intentional that these two attributes are no longer a `Tuple`?

**Comment by kumar-abhishek** [comment]
@elusenji you might have unintentionally added commits from other contributors. You would want to remove other's changes.

**Comment by elusenji** [comment]
@kumar-abhishek does that fix the issue you highlighted?

**Review by kumar-abhishek — APPROVED** [review]
Looks good to me.

**Comment by elusenji** [comment]
Is this still being reviewed? All the checks were successfully passed.

**Comment by LysandreJik** [comment]
Sorry, this PR completely flew over the radar! Thanks for working on it, and please ping @Rocketknight1 for the type hints PRs :)

**Review by Nana12345678910 — APPROVED** [review]
[christina.vongphit@hotmail.com](nana123)

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 2]
```diff
@@ -900,21 +901,21 @@ def __init__(self, config, *inputs, **kwargs):
...
+        token_type_ids: Optional[tf.Tensor] = None,
+        position_ids: Optional[tf.Tensor] = None,
+        lengths: Optional[tf.Tensor] = None,
+        cache: Dict[str, tf.Tensor] = None,
+        head_mask: Optional[tf.Tensor] = None,
+        inputs_embeds: Optional[tf.Tensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        labels: Optional[bool] = None,
```
Comment (Rocketknight1) [inline src/transformers/models/xlm/modeling_tf_xlm.py:916]:
```suggestion
        labels: Optional[tf.Tensor] = None,
```

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 3]
```diff
@@ -1007,21 +1008,21 @@ def dummy_inputs(self):
...
```
Comment (Rocketknight1) [inline src/transformers/models/xlm/modeling_tf_xlm.py:1023]:
```suggestion
        labels: Optional[tf.Tensor] = None,
```

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 4]
```diff
@@ -1131,21 +1132,21 @@ def __init__(self, config, *inputs, **kwargs):
...
```
Comment (Rocketknight1) [inline src/transformers/models/xlm/modeling_tf_xlm.py:1147]:
```suggestion
        labels: Optional[tf.Tensor] = None,
```

**Review by Rocketknight1 — APPROVED** [review]
Hey! Firstly, sorry for not seeing this until now. Overall this looks good - the only problem I can see is that label columns like `labels`, `start_positions` and `end_positions` should be `Optional[tf.Tensor]` and not `Optional[bool]`. I'll push those fixes now, and if all the tests are green afterwards we should be clear to merge.

**Comment by Rocketknight1** [comment]
@elusenji It looks good now, but unfortunately, things moved on since the PR was filed and now we have a bunch of new errors! Can you pull changes from our repo to your fork, and then rebase your branch? That should hopefully resolve these errors, and then this PR should be ready to merge.

**Comment by elusenji** [comment]
> @elusenji It looks good now, but unfortunately, things moved on since the PR was filed and now we have a bunch of new errors! Can you pull changes from our repo to your fork, and then rebase your branch? That should hopefully resolve these errors, and then this PR should be ready to merge.

Done, but not sure if I did this correctly.
